### PR TITLE
OSDOCS-2446: Updating tested firewall info and adding content from OCP for one-sto…

### PR DESCRIPTION
**Pull request by jonquilwilliams**
…p shopping.

JIRA: https://issues.redhat.com/browse/OSDOCS-2446

Preview: https://deploy-preview-35971--osdocs.netlify.app/openshift-rosa/latest/rosa_getting_started/rosa-aws-prereqs?utm_source=github&utm_campaign=bot_dp#osd-aws-privatelink-firewall-prerequisites

Branch = dedicated-4. No CP/milestones.

### CODE REVIEW

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc [hunk 1]
```diff
@@ -1,77 +1,118 @@
...
+|Required. The `https://cloud.redhat.com/openshift` site uses authentication from `sso.redhat.com` to  download the pull secret and use Red Hat SaaS solutions to facilitate monitoring of your subscriptions, cluster inventory, chargeback reporting, and so on.
+|443, 80
 
 |`pull.q1w2.quay.rhcloud.com`
-|Provides a fallback registry used by the cluster when quay.io is not available.
-|Recommended
+|Recommended. Provides a fallback registry used by the cluster when quay.io is not available.
+|443, 80
+
+|`*.q1w2.quay.rhcloud.com`
```
Comment (dofinn) [inline modules/osd-aws-privatelink-firewall-prerequisites.adoc:37]:
```suggestion
|`.q1w2.quay.rhcloud.com`
```

As per AWS wildcard documentation.

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc [hunk 2]
```diff
@@ -1,77 +1,118 @@
...
+|443, 80
+
+|`quay-registry.s3.amazonaws.com`
+|Required. Used to access Quay image content in AWS.
+|443, 80
 |===
 +
 When you add a site such as `quay.io` to your allowlist, do not add a wildcard entry such as `*.quay.io` to your denylist. In most cases, image registries use a content delivery network (CDN) to serve images. If a firewall blocks access, then image downloads are denied when the initial download request is redirected to a host name such as `cdn01.quay.io`.
++
+CDN host names, such as `cdn01.quay.io`, are covered when you add a wildcard entry, such as ``*.quay.io`, in your allowlist.
```
Comment (dofinn) [inline modules/osd-aws-privatelink-firewall-prerequisites.adoc:56]:
```suggestion
CDN host names, such as `cdn01.quay.io`, are covered when you add a wildcard entry, such as ``.quay.io`, in your allowlist.
```

Given this doc is strictly AWS specific, lets be consistent with the expectations of how to configure a wildcard?

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc [hunk 3]
```diff
@@ -1,77 +1,118 @@
...
-|Required to access Amazon Elastic Load Balancers (ELBs) for API and application load balancing.
-|Required
+|Required. Used to access Amazon Elastic Load Balancers (ELBs) for API and application load balancing.
+|443, 80
 
 |`<cluster_id>-<shard>.<aws_region>.amazonaws.com`
-|Required for access to the registry. Replace `<cluster_id>`, `<shard>`, and `<aws_region>` with your cluster ID, shard name and AWS region code.
-
-|Required
+|Required. Used for access to the registry. Replace `<cluster_id>`, `<shard>`, and `<aws_region>` with your cluster ID, shard name and AWS region code.
```
Comment (dofinn) [inline modules/osd-aws-privatelink-firewall-prerequisites.adoc:114]: we should explicity call out that these details wont be available until post/during installation? and how to actually get them?

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc [hunk 4]
```diff
@@ -83,65 +124,155 @@ Review the link:https://docs.aws.amazon.com/general/latest/gr/rande.html[AWS Ser
...
+|Used to access mirrored installation content and images. This site is also a source of release image signatures, although the Cluster Version Operator needs only a single functioning source. Required if you do not allow `storage.googleapis.com/openshift-release`.
+|443, 80
 
 |`storage.googleapis.com/openshift-release`
-|Alternative site to download platform release signatures that are used by the cluster to know what images to pull from quay.io.
-|Recommended
+|Recommended. Alternative site to mirror.openshift.com/. Used to download platform release signatures that are used by the cluster to know what images to pull from quay.io.
+|443, 80
+
+|`*.apps.<cluster_name>.<base_domain>`
```
Comment (dofinn) [inline modules/osd-aws-privatelink-firewall-prerequisites.adoc:139]:
```suggestion
|`.apps.<cluster_name>.<base_domain>`
```

as per previous wildcard comments

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc [hunk 5]
```diff
@@ -83,65 +124,155 @@ Review the link:https://docs.aws.amazon.com/general/latest/gr/rande.html[AWS Ser
...
+
+|`route53.amazonaws.com`
+|Required. Used to install and manage clusters in an AWS environment.
+|443, 80
+
+|`sts.amazonaws.com`
+|Required. Used to install and manage clusters in an AWS environment.
+|443, 80
+
+|`ec2.us-east-1.amazonaws.com`
```
Comment (dofinn) [inline modules/osd-aws-privatelink-firewall-prerequisites.adoc:191]:
```suggestion
|`ec2.<aws-region>.amazonaws.com`
```

Keep it consistent with rest of doc for any variables?

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc [hunk 6]
```diff
@@ -83,65 +124,155 @@ Review the link:https://docs.aws.amazon.com/general/latest/gr/rande.html[AWS Ser
...
+|Required. Region dependent and must be added per cluster and per region.
+|443, 80
+
+|`CLUSTER-NAME-k5bxz-image-registry-us-east-1-lsiflffxtmfyikx.s3.dualstack.us-east-1.amazonaws.com`
```
Comment (dofinn) [inline modules/osd-aws-privatelink-firewall-prerequisites.adoc:195]:
```suggestion
|`CLUSTER-NAME-k5bxz-image-registry-us-east-1-lsiflffxtmfyikx.s3.dualstack.us-east-1.amazonaws.com`
```

This is also created at install time, customer will need to retrieve their endpoint via:

```
oc -n openshift-image-registry get pod -l docker-registry=default -o json | jq '.items[].spec.containers[].env[] | select(.name=="REGISTRY_STORAGE_S3_BUCKET")'
```

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc [hunk 7]
```diff
@@ -83,65 +124,155 @@ Review the link:https://docs.aws.amazon.com/general/latest/gr/rande.html[AWS Ser
...
+|443, 80
+|===
+
+. Operators require route access to perform health checks. Specifically, the authentication and web console Operators connect to two routes to verify that the routes work. If you are the cluster administrator and do not want to allow *.apps.<cluster_name>.<base_domain>, then you must allow these routes:
++
+[cols="3,4,1",options="header"]
+|===
+|URL | Function | Port
+
+|`oauth-openshift.apps.<cluster_name>.<base_domain>`
```
Comment (dofinn) [inline modules/osd-aws-privatelink-firewall-prerequisites.adoc:210]: missing <shard> as referenced in other routes above

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc [hunk 8]
```diff
@@ -83,65 +124,155 @@ Review the link:https://docs.aws.amazon.com/general/latest/gr/rande.html[AWS Ser
...
+|Required.
+|443
+
+|`console-openshift-console.apps.<cluster_name>.<base_domain>`, or the host name that is specified in the `spec.route.hostname` field of the `consoles.operator/cluster` object if the field is not empty
```
Comment (dofinn) [inline modules/osd-aws-privatelink-firewall-prerequisites.adoc:214]: missing <shard> as referenced in other routes above

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc [hunk 9]
```diff
@@ -83,65 +124,155 @@ Review the link:https://docs.aws.amazon.com/general/latest/gr/rande.html[AWS Ser
...
+|Required. Alerting service used by OpenShift Dedicated to send periodic pings that indicate whether the cluster is available and running.
+|443
 
 |`sftp.access.redhat.com`
-|FTP server used by must-gather-operator to upload diagnostic logs to help troubleshoot issues with the cluster.
-|Recommended
+|Recommended. FTP server used by must-gather-operator to upload diagnostic logs to help troubleshoot issues with the cluster.
+|443
 
 |`*.osdsecuritylogs.splunkcloud.com`
```
Comment (dofinn) [inline modules/osd-aws-privatelink-firewall-prerequisites.adoc:252]:
```suggestion
|`.osdsecuritylogs.splunkcloud.com`
```

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc (hunk 3 above)
Comment (jonquilwilliams) [reply modules/osd-aws-privatelink-firewall-prerequisites.adoc:114]: @dofinn  How do they get them?

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc (hunk 6 above)
Comment (jonquilwilliams) [reply modules/osd-aws-privatelink-firewall-prerequisites.adoc:195]: I don;t see a difference in the lines, but assume you want me to replace `us-east-1` with <aws_region>, right?

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc (hunk 3 above)
Comment (jonquilwilliams) [reply modules/osd-aws-privatelink-firewall-prerequisites.adoc:114]: Please see if you agree with what I've drafted. Thanks.

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc (hunk 6 above)
Comment (dofinn) [reply modules/osd-aws-privatelink-firewall-prerequisites.adoc]: This endpoint is generated dynamically during installation like the shard etc. We should call this out to customer and advise how to retrieve it :)

```
[~ {stg} (dofinn:default)]$ oc -n openshift-image-registry get pod -l docker-registry=default -o json | jq '.items[].spec.containers[].env[] | select(.name=="REGISTRY_STORAGE_S3_BUCKET")'

{
  "name": "REGISTRY_STORAGE_S3_BUCKET",
  "value": "dofinn-tm5zn-image-registry-us-west-2-lxomiuyebfutxsfl"
}
{
  "name": "REGISTRY_STORAGE_S3_BUCKET",
  "value": "dofinn-tm5zn-image-registry-us-west-2-lxomiuyebfutxsfl"
}
```

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc (hunk 3 above)
Comment (dofinn) [reply modules/osd-aws-privatelink-firewall-prerequisites.adoc]: /lgtm

**Comment by yuwang-RH** [comment]
LGTM

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc [hunk 10]
```diff
@@ -1,147 +1,291 @@
...
 +
 When you add a site such as `quay.io` to your allowlist, do not add a wildcard entry such as `*.quay.io` to your denylist. In most cases, image registries use a content delivery network (CDN) to serve images. If a firewall blocks access, then image downloads are denied when the initial download request is redirected to a host name such as `cdn01.quay.io`.
++
+CDN host names, such as `cdn01.quay.io`, are covered when you add a wildcard entry, such as `.quay.io`, in your allowlist.
++
+Allowlist any site that provides resources for a language or framework that your builds require.

 
-. Managed clusters require Telemetry to be enabled to allow Red Hat to react more quickly to problems and better support our customers, as well as better understand how product upgrades impact clusters.  
+. Managed clusters require telemetry to be enabled to allow Red Hat to react more quickly to problems and better support our customers, and to better understand how product upgrades impact clusters.
```
Comment (skrthomas) [inline modules/osd-aws-privatelink-firewall-prerequisites.adoc:60]: You don't need a comma here.

**Review by skrthomas — COMMENTED** [review]
Just a small lil comment. Great work :)

**Pull request merged**